public/pages/workflow_detail/agentic_search/configure_flow/mcp_tool_filters.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect } from 'react';
+import { getIn } from 'formik';
+import { EuiComboBox } from '@elastic/eui';
+import { MCPConnector } from '../../../../../common';
+
+interface MCPToolFiltersProps {
+  mcpServer: MCPConnector;
+  index: number;
+  updateMCPServer: (mcpServer: MCPConnector, index: number) => void;
+}
+
+export function MCPToolFilters(props: MCPToolFiltersProps) {
+  const [selectedFilters, setSelectedFilters] = useState<string[]>([]);
+  useEffect(() => {
+    setSelectedFilters(getIn(props, 'mcpServer.tool_filters', []) as string[]);
+  }, [props.mcpServer]);
+
+  function handleOptionsChange(updatedFilters: string[]): void {
+    props.updateMCPServer(
+      {
+        ...props.mcpServer,
+        tool_filters: updatedFilters,
+      },
+      props.index
+    );
+  }
+
+  return (
+    <EuiComboBox
+      noSuggestions
+      compressed
+      placeholder="Create some filters"
+      selectedOptions={selectedFilters.map((selectedFilter) => {
+        return {
+          label: selectedFilter,
+          value: selectedFilter,
+        };
+      })}
+      onChange={(options) => {
+        handleOptionsChange(options.map((option) => option.value) as string[]);
+      }}
+      onCreateOption={(newFilter) =>
+        handleOptionsChange([...selectedFilters, newFilter])
```

Comment:
Should we validate the new filter?

Reply:
I don't believe it's needed. The component itself enforces that it is a non-empty string. And for now, I think it's best not to limit the string length for maximum flexibility (which could be overridden in JSON editor directly anyways).